cmd/dagger/main.go

```diff
@@ -345,6 +346,34 @@ func toUpperBold(s string) string {
 	return termenv.String(upperCase).Bold().String()
 }
 
+// separateAndModifyFlags separates optional flags from required flags
+func separateAndModifyFlags(originalFlags *pflag.FlagSet) *pflag.FlagSet {
+	mergedFlags := pflag.NewFlagSet("merged", pflag.ContinueOnError)
+	mergedFlags.SortFlags = false
+
+	optionalFlags := pflag.NewFlagSet("optional", pflag.ContinueOnError)
+
+	// separate optional flags from required flags
+	originalFlags.VisitAll(func(flag *pflag.Flag) {
+		clone := *flag
```

Comment:
This isn’t enough. It’ll make a shallow clone of the `Flag` struct but will leave out the annotations map and you need it for validation. Even if you then also manually clone the annotations, it’s possible in a later update a slice field is added to `Flag`. I think it’s ok to change the flag here without cloning.

Another alternative would be to append `[required]` only to function arguments, while creating the flags for them as I don’t expect we’ll use required flags anywhere else.